### ISSUE TITLE
Editing messages

### ISSUE THREAD

**Issue by qxuken** [telegram-bot-rb/telegram-bot#36]
Is there any better way to edit messages inside the controller than that?

`Telegram.bot.edit_message_text(chat_id: from['id'], text: "edited message", message_id: payload['message']['message_id'])`

**Comment by printercu** [comment]
Nope, but it's easy to implement. I've used this methods in a project:

```ruby
    def edit_message(type, params = {})
      params =
        if payload['inline_message_id']
          params.merge(inline_query_id: payload['inline_message_id'])
        elsif message = payload['message'] # rubocop:disable AssignmentInCondition
          params.merge(chat_id: message['chat']['id'], message_id: message['message_id'])
        else
          raise 'Can not edit message without `inline_message_id` or `message`'
        end
      edit_message_safely(type, params)
      answer_callback_query params[:text] if type == :text && params[:text]
    end

    # Silently ignores exceptions when message is not modified.
    def edit_message_safely(type, params)
      bot.public_send("edit_message_#{type}", params)
    rescue Telegram::Bot::Error => e
      raise e unless e.message.include?('message is not modified')
    end

    # Silently ignore old queries.
    def answer_callback_query(*)
      super
    rescue Telegram::Bot::Error => e
      raise e unless e.message.include?('QUERY_ID_INVALID')
    end
```

Use it like this: 
```ruby
edit_message :reply_markup, reply_markup: new_keyboard
edit_message(:text, text:'new text')
```

**Comment by qxuken** [comment]
Works perfectly. Good feature to implement into the gem.

**Comment by printercu** [comment]
Added `edit_message` helper in https://github.com/telegram-bot-rb/telegram-bot/commit/6879618c004bf563c2b61db8eec8333bddcd1dfd Will release in few days.

I haven't add that rescues from example above to gem because they are very app-specific. But I've collected them on new page in wiki: 
https://github.com/telegram-bot-rb/telegram-bot/wiki/Basic-workarounds

**Comment by printercu** [comment]
Available in 0.12
